crates/bevy_pbr/src/meshlet/asset.rs

```diff
@@ -0,0 +1,103 @@
+use bevy_asset::{
+    io::{Reader, Writer},
+    saver::{AssetSaver, SavedAsset},
+    Asset, AssetLoader, AsyncReadExt, AsyncWriteExt, LoadContext,
+};
+use bevy_math::Vec3;
+use bevy_reflect::TypePath;
+use bevy_utils::BoxedFuture;
+use bytemuck::{Pod, Zeroable};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+/// A mesh that has been pre-processed into multiple small clusters of triangles called meshlets.
+///
+/// There are restrictions on the [`crate::Material`] functionality that can be used with this type of mesh.
+/// * Materials have no control over the vertex shader, and can only have a custom fragment shader.
+/// * Materials must be opaque. Transparent, alpha masked, and transmissive materials are not supported.
+/// * Materials must use the [`crate::Material::meshlet_mesh_fragment_shader`] method,
+///   which requires certain shader patterns that differ from the regular material shaders.
+/// * Limited control over [`bevy_render::render_resource::RenderPipelineDescriptor`] attributes.
+///
+/// See also [`super::MaterialMeshletMeshBundle`].
+#[derive(Asset, TypePath, Serialize, Deserialize, Clone)]
+pub struct MeshletMesh {
+    /// The total amount of indices in the overall mesh (sum of meshlet index_counts).
+    pub total_meshlet_indices: u64,
+    /// Raw vertex data bytes for the overall mesh.
+    pub vertex_data: Arc<[u8]>,
+    /// Indices into `vertex_data`.
+    pub vertex_ids: Arc<[u32]>,
+    /// Indices into `vertex_ids`.
+    pub indices: Arc<[u8]>,
+    /// The list of meshlets making up this mesh.
+    pub meshlets: Arc<[Meshlet]>,
+    /// A list of spherical bounding volumes, 1 per meshlet.
+    pub meshlet_bounding_spheres: Arc<[MeshletBoundingSphere]>,
```

Comment:
It's easier if it's kept separate, as it makes uploading the data to the GPU easier.

Reply:
I'm guessing what I've missed is that some of the meshlet passes need only the meshlet ranges or the bounding spheres?

Reply:
Yes. On the CPU, I chose to keep them separate because they're going to be separate on the GPU anyways. On the GPU, having them separate makes a lot of sense. Only the culling passes need the bounding sphere data, and due to memory alignment and cache concerns, it's better to have them in separate buffers.